crates/core/src/view/search_bar.rs

```diff
@@ -20,55 +20,77 @@ pub struct SearchBar {
 }
 
 impl SearchBar {
-    pub fn new(rect: Rectangle, input_id: ViewId, placeholder: &str, text: &str, context: &mut Context) -> SearchBar {
-        let id = ID_FEEDER.next();
-        let mut children = Vec::new();
+    pub fn new_empty(availible_space: Rectangle) -> SearchBar {
         let dpi = CURRENT_DEVICE.dpi;
+        let small_height = scale_by_dpi(SMALL_BAR_HEIGHT, dpi) as i32;
         let thickness = scale_by_dpi(THICKNESS_MEDIUM, dpi) as i32;
-        let side = rect.height() as i32;
+        let bar_height = small_height + thickness;
```

Comment:
SearchBar now determines it's own size based on available space, so that it is consistently sized wherever it is used